core/graphelier-service/api/hndlrs/playbackhandler.go

```diff
@@ -0,0 +1,248 @@
+package hndlrs
+
+import (
+	"graphelier/core/graphelier-service/db"
+	"graphelier/core/graphelier-service/models"
+	"net"
+	"net/http"
+	"strconv"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/gorilla/mux"
+	"github.com/gorilla/websocket"
+)
+
+// SocketConn : Indirection layer for access to websocket.Conn
+type SocketConn interface {
+	RemoteAddr() net.Addr
+	WriteJSON(interface{}) error
+	ReadMessage() (int, []byte, error)
+	Close() error
+}
+
+// PlaybackSession : Handles lifetime of a playback session with the client over a websocket connection
+type PlaybackSession struct {
+	Socket    SocketConn
+	Orderbook *models.Orderbook
+	Delay     uint64
+	Loader    MessageLoader
+	messages  chan []*models.Message
+	realtime  time.Time
+	running   bool
+}
+
+// MessageLoader : Interface for asynchronous retrieval of messages, each call to LoadMessages should be stateful and move the bounds
+type MessageLoader interface {
+	LoadMessages()
+	Init(*models.Orderbook, chan []*models.Message)
+}
+
+// TimeIntervalLoader : Loads messages by time intervals
+type TimeIntervalLoader struct {
+	Instrument       string
+	Datastore        db.Datastore
+	Interval         uint64
+	messages         chan []*models.Message
+	currentTimestamp uint64
+}
+
+// CountIntervalLoader : Loads messages by regular count intervals
+type CountIntervalLoader struct {
+	Instrument  string
+	Datastore   db.Datastore
+	Count       uint64
+	messages    chan []*models.Message
+	currentPage models.Paginator
+}
+
+var upgrader = websocket.Upgrader{}
+
+// StreamPlayback : Initiates a websocket connection and starts streaming order book modifications for playback
+func StreamPlayback(env *Env, w http.ResponseWriter, r *http.Request) error {
```

Comment:
Does this handle the case where a user switches the playback settings while the playback is occurring?

Reply:
No, so far this year case would be handled by stopping the session and starting a new one though if we find there is value in handling it in the same session then it should be do-able